src/app/mesh/qgsmeshcalculatordialog.h

```diff
@@ -34,15 +34,19 @@ class APP_EXPORT QgsMeshCalculatorDialog: public QDialog, private Ui::QgsMeshCal
      * \param parent widget
      * \param f window flags
      */
-    QgsMeshCalculatorDialog( QgsMeshLayer *meshLayer = nullptr, QWidget *parent = nullptr, Qt::WindowFlags f = Qt::WindowFlags() );
+    QgsMeshCalculatorDialog(
+      const QgsRectangle &currentExtent,
+      const QgsCoordinateReferenceSystem &currentCrs,
+      QgsMeshLayer *meshLayer = nullptr,
+      QWidget *parent = nullptr,
+      Qt::WindowFlags f = Qt::WindowFlags() );
```

Comment:
How about:

```suggestion
    QgsMeshCalculatorDialog(
      QgsMeshLayer *meshLayer = nullptr,
      QgsMapCanvas* mapCanvas = nullptr,
      QWidget *parent = nullptr,
      Qt::WindowFlags f = Qt::WindowFlags() );
```

instead, and then retrieving the extent/crs from the canvas instead the dialog constructor. That's a cleaner approach in my view (also for the raster calculator)

Reply:
@nyalldawson done! Thanks for the review